src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
```

Comment:
I suggest having two methods on `Tmdb`, like `.guestSession()` and `.accountSession(username, password)`. Whichever was called last determines the current auth state.

Then the authenticator and interceptor should get the mode and credentials from `Tmdb` and call the appropriate auth methods and/or add the appropriate credentials.